server/coordinator/procedure/ddl/droppartitiontable/drop_partition_table.go

```diff
@@ -278,10 +282,21 @@ func dropDataTablesCallback(event *fsm.Event) {
 
 	shardVersions := req.p.relatedVersionInfo.ShardWithVersion
 	for _, tableName := range params.SourceReq.PartitionTableInfo.SubTableNames {
-		table, shardVersionUpdate, err := ddl.GetShardVersionByTableName(params.ClusterMetadata, req.schemaName(), tableName, shardVersions)
+		table, err := ddl.GetTableMetadata(params.ClusterMetadata, req.schemaName(), tableName)
 		if err != nil {
-			procedure.CancelEventWithLog(event, err, fmt.Sprintf("get shard version by table, table:%s", tableName))
-			return
+			log.Warn("get table metadata failed", zap.String("tableName", tableName))
+			continue
+		}
+
+		shardVersionUpdate, err := ddl.GetTableShardVersion(table, params.ClusterMetadata, shardVersions)
+		if err != nil {
```

Comment:
Not every error thrown by the `ddl.GetTableShardVersion` can allow us to remove the table metadata, maybe we should detect the error that the shard can't be found, and only do removal of table metadata for this error.